package.json

```diff
@@ -57,8 +57,9 @@
     "xy-parser": "^3.0.0"
   },
   "dependencies": {
-    "ml-optimize-lorentzian": "0.1.4",
-    "ml-savitzky-golay-generalized": "2.0.1"
+    "ml-optimize-lorentzian": "^0.1.5",
+    "ml-savitzky-golay-generalized": "2.0.1",
+    "ml-spectra-processing": "^3.0.2"
```

Comment:
I would not use this specific noise determination because it has some heavy dependencies and is known to work well for NMR but this tool is use in infrared, mass, chromatography, etc.
You can always determine the noise before and give it as a parameter.
The noise calculation use should nevertheless be in ml-spectra-processing
Median is a possibility for noise determination. Don't know exactly what is the current algorithm.